.eslintrc.json

```diff
@@ -0,0 +1,341 @@
+{
```

Comment:
@makarandp0 Can we indent with 2 spaces? Also, can you make sure that we retain only those rules present in the `.eslintrc` file? That way we don't have to make any (or very little) modifications in the source code.

Reply:
@manjeshbhargav - I will update with 2 spaces indentation for this json file. 


About retaining previous rule - I have ensured that all previous rules are carried over. 
Upgrading to newer eslint gives us an opportunity to catch more issues that were not in older version. Looks like our code is mostly clean, and it required only few exceptions that too mostly in test code.  I think it would be good to keep new checks by default, and remove new rules on case-by-case basis as we find them bothersome. Right now I have not noticed any that deserves blocking.  

Yes down side is some more files are modified in this PR, but right now they are all done, and it's more work to remove those changes.